### ISSUE TITLE
Reading : BERT Pre-training of Deep Bidirectional Transformers for Language Understanding

### ISSUE THREAD

**Issue by tikiti-kyotaro** [tikiti-kyotaro/Paperlog#31]
# Abstract
提案モデル： BERT
BERTの利点：層を一つ加えるだけでファインチューニングができ、幅広いタスクの最先端のものを実現することができる⇨それぞれのタスクに合わせて大幅なアーキテクチャの改変を必要としない
11のタスクに対して最先端の結果を得ることができた

# Introduction
事前学習は文単位でもトークン単位でも役に立つ
既存の事前学習をダウンストリームタスクに適応させる技術
１、feature-based
２、fine-tuning

feature-basedは事前学習された表現を追加の特徴料として扱い、モデルに特化したアーキテクチャを構成する。
fine-tuningはモデルに依存するパラメータは少ない。全ての事前学習パラメータを微調整することで適応する

事前学習中に目的関数を共有して、単一方向言語モデルを用いて一般的な言語への拡張を謀る

ただ、これらの手法は事前学習の能力をフルで発揮できてはいない。
要因：単言語モデルは一方向であるため、事前学習時のアーキテクチャが限られるということ

OpenAI GPTは左から右という一方向、Transformerは全てのトークンにおいて、そのトークンより前の部分しか見ていないものであったため、ファインチューニングできた。

ただ、文単位のタスクや両方向の文脈が必要なものにとっては向いていない（というか有害な可能性あり）

提案タスクのBERTを使ってファインチューニングを改善しよう！

BERTはMLMというものの目的関数を用いて単一方向の制約を減らす

MLM (masked laungage model) : 入力されたトークンのいくつかをランダムに隠す。
目的はその隠されたトークンの単語IDを予測することである。MASKを適切に予測できるようにパラメータを更新することによって文の文脈を考慮できるようにする
→これを使うことで左右の文脈を融合することができる

NSP (next sentence prediction) : ２つの文章を与えて、それらが隣り合っているかいないかを２値判定する。２つの文の一部をそれぞれMASKして、その文につながりがあるかないかということを２値分類するようなタスク
→これによってMLMだけではできなかった文単位の処理を可能にしている。

二つのモデルの解説：
https://ainow.ai/2019/05/08/166723/#Next_Sentence_Prediction

## Contibution
１、双方向の事前学習の重要さ　BERTなら双方向の事前学習をMLMを用いて可能にする
２、タスクに固有のアーキテクチャを開発しなくても良い
３、BERTは11個ものタスクのsotaを進めた

# Related Work
## Unsupervised feature based
一旦飛ばす（今回はFine tuningのため）

## Unsupervised fine tuning
元は単語から埋め込みベクトルへどのように変換するのかということだけを事前学習
最近になって進化していきた→文脈に応じたトークンを表現するエンコーダがダウンストリームタスクに応じて微調整するような手法
少ないパラメータでいけるのが利点（学習をするときにパラメータを一から学習しなくて済む）

## Transfer learning
大規模なデータを用いた教師あり学習を他のものに移す

# Model
事前学習とファインチューニングの２つのステップを組み合わせたものがBERTである
Figure1
１、まず正解ラベルがないマスクされた文を使って学習（パラメータ更新）
２、その事前学習された結果をもとに正解ラベルがついているデータを用いてタスクごとにファインチューニングする

BERTの利点：

# Concludion
unsurprisedの事前学習はとても大事
同じ事前学習モデルで広範囲のタスクに対して対応できるようになったのが成果である
